cmd/sync/model.go

```diff
@@ -8,27 +8,50 @@ package main
 type SyncConfigEntry struct {
 	// Upstream is the upstream Git repository to take updates from.
 	Upstream string
+
+	// UpstreamMirror is an optional upstream-maintained mirror of the Upstream repository.
+	// Specifically, for Upstream 'https://go.googlesource.com/go', the UpstreamMirror is
+	// 'https://github.com/golang/go'.
+	//
+	// When updating a submodule, sync checks both repos and only updates to a version that is in
+	// both. This ensures our dev process doesn't have a strong dependency on one copy of upstream's
+	// code or the other, and makes it reasonable to point the submodule at the GitHub mirror of Go
+	// by default, which has a better appearance in the GitHub UI for a submodule: a clickable link.
+	//
+	// When merging a fork, the check is skipped. A fork inherently mirrors the upstream code, so
+	// there is no reason to look for a common version and hold back.
+	//
+	// If UpstreamMirror is not defined (default), the update simply uses the latest version of
+	// Upstream.
+	UpstreamMirror string
+
```

Comment:
Super nit: some fields have blank lines between them, others don't. OCD jared is satsified just writing that down.

Reply:
The idea was they're in groups ("stuff we fetch from", "stuff we push to", "branches", "how is the update carried out"). But a newline snuck in before the `UpstreamMirror` doc comment.

In theory, could use struct embedding to separate these with actual Go syntax (giving us a place to write actual names for these categories)... but that would make it harder to look through the JSON file and correlate each key with one of these fields.

Reply:
For now, removed the empty line above `UpstreamMirror` to make this fit the rough categorization I was going for. Not sure if it's worth doing more to point out the categories, or just make it consistent to avoid questions.